lib/Dashboard.js

```diff
@@ -0,0 +1,54 @@
+var rp = require('./request-promise'),
+  config = require('../config'),
+  _ = require('lodash'),
+  Q = require('q');
+
+var Module = require('./Module');
+
+function Dashboard (slug) {
+  this.options = {
+    json: true,
+    slug: slug
+  };
+}
+
+Dashboard.prototype.getConfig = function () {
+  var options = _.extend(this.options, {
+    url: config.stagecraft + 'public/dashboards?slug=' + this.options.slug
+  });
+
+  return rp(options);
+};
+
+Dashboard.prototype.resolve = function () {
```

Comment:
I wanted a standard name for "do everything" that could be used through the dashboard and module, modules. I was also hoping that as a user you didn't care too much about what's going on under the hood. I'd be interested to see what other peoples thoughts are on this? I'm open to suggestions around this? Do you have any thoughts on what it could be renamed too?